tests/Support/SupportStrTest.php

```diff
@@ -216,6 +216,25 @@ public function testStrBefore()
         $this->assertSame('han', Str::before('han0nah', '0'));
         $this->assertSame('han', Str::before('han0nah', 0));
         $this->assertSame('han', Str::before('han2nah', 2));
+        $this->assertSame('hannah', Str::before('hannah', 'NAH'));
```

Comment:
Added a case sensitive to the previous functionality to show that it does not match